src/app/shared/account/provider.service.spec.ts

```diff
@@ -1,71 +1,46 @@
-import { inject, TestBed } from '@angular/core/testing';
-import { BaseRequestOptions, Http } from '@angular/http';
-import { MockBackend } from '@angular/http/testing';
+import { TestBed } from '@angular/core/testing';
+
+import {
+  HttpClientTestingModule,
+  HttpTestingController
+} from '@angular/common/http/testing';
 
 import { Broadcaster, Logger } from 'ngx-base';
 import { AUTH_API_URL, AuthenticationService } from 'ngx-login-client';
 import { ProviderService } from './provider.service';
 
-describe('Service: Provider Service', () => {
+import { createMock } from 'testing/mock';
 
-    let mockService: MockBackend;
-    let providerService: ProviderService;
-    let broadcaster: Broadcaster;
-    let fakeAuthService: any;
+describe('Service: Provider Service', () => {
 
+    let service: ProviderService;
+    let controller: HttpTestingController;
+    let mockLogger: jasmine.SpyObj<Logger>;
     class BroadcasterTestProvider {
       static broadcaster = new Broadcaster();
     }
 
     beforeEach(() => {
-        fakeAuthService = {
-            getToken: function() {
-              return 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiY2xpZW50X3Nlc3Npb24iOiJURVNUU0VTU0lPTiIsInNlc3Npb25fc3RhdGUiOiJURVNUU0VTU0lPTlNUQVRFIiwiYWRtaW4iOnRydWUsImp0aSI6ImY5NWQyNmZlLWFkYzgtNDc0YS05MTk0LWRjM2E0YWFiYzUwMiIsImlhdCI6MTUxMDU3MTMxOSwiZXhwIjoxNTEwNTgwODI3fQ.l0m6EFvk5jbND3VOXL3gTkzTz0lYQtPtXS_6C24kPQk';
-            },
-            isLoggedIn: function() {
-              return true;
-            }
-          };
-
-        TestBed.configureTestingModule({
-          providers: [
-            BaseRequestOptions,
-            ProviderService,
-            MockBackend,
-            {
-              provide: Http,
-              useFactory: (backend: MockBackend,
-                           options: BaseRequestOptions) => new Http(backend, options),
-              deps: [MockBackend, BaseRequestOptions]
-            },
-            {
-              provide: AUTH_API_URL,
-              useValue: 'https://auth.fabric8.io/api/'
-            },
-            {
-                provide: AuthenticationService,
-                useValue: fakeAuthService
-            },
-            {
-              provide: Broadcaster,
-              useValue: BroadcasterTestProvider.broadcaster
-            },
-            Logger
-          ]
-        });
+      const mockAuthenticationService: jasmine.SpyObj<AuthenticationService> = createMock(AuthenticationService);
+      mockAuthenticationService.getToken.and.returnValue('eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiY2xpZW50X3Nlc3Npb24iOiJURVNUU0VTU0lPTiIsInNlc3Npb25fc3RhdGUiOiJURVNUU0VTU0lPTlNUQVRFIiwiYWRtaW4iOnRydWUsImp0aSI6ImY5NWQyNmZlLWFkYzgtNDc0YS05MTk0LWRjM2E0YWFiYzUwMiIsImlhdCI6MTUxMDU3MTMxOSwiZXhwIjoxNTEwNTgwODI3fQ.l0m6EFvk5jbND3VOXL3gTkzTz0lYQtPtXS_6C24kPQk');
```

Comment:
Can we replace this token string with something a little more obviously mocked, like `mock-auth-token`, while we're here?

Reply:
This token actually contains information which is decoded using `jwt-decode` to form the URL.

Reply:
Done.